src/sentry/identity/vsts/provider.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import absolute_import
+
+from django import forms
+
+from sentry import http
+from sentry import options
+
+from sentry.web.helpers import render_to_response
+from sentry.options.manager import FLAG_PRIORITIZE_DISK
+from sentry.identity.oauth2 import OAuth2Provider, OAuth2LoginView, OAuth2CallbackView
+from sentry.pipeline import PipelineView
+
+options.register('vsts.client-id', flags=FLAG_PRIORITIZE_DISK)
+options.register('vsts.client-secret', flags=FLAG_PRIORITIZE_DISK)
+options.register('vsts.verification-token', flags=FLAG_PRIORITIZE_DISK)
+
+
+class VSTSIdentityProvider(OAuth2Provider):
+    key = 'vsts'
+    name = 'Visual Studio Team Services'
+
+    oauth_access_token_url = 'https://app.vssps.visualstudio.com/oauth2/token'
+    oauth_authorize_url = 'https://app.vssps.visualstudio.com/oauth2/authorize'
+
+    oauth_scopes = (
+        'vso.build_execute',
+        'vso.code_full',
+        'vso.codesearch',
+        'vso.connected_server',
+        'vso.dashboards_manage',
+        'vso.entitlements',
+        'vso.extension.data_write',
+        'vso.extension_manage',
+        'vso.gallery_manage',
+        'vso.graph_manage',
+        'vso.identity_manage',
+        'vso.loadtest',
+        'vso.machinegroup_manage',
+        'vso.memberentitlementmanagement_write',
+        'vso.notification_diagnostics',
+        'vso.notification_manage',
+        'vso.packaging_manage',
+        'vso.profile_write',
+        'vso.project_manage',
+        'vso.release_manage',
+        'vso.security_manage',
+        'vso.serviceendpoint_manage',
+        'vso.symbols_manage',
+        'vso.taskgroups_manage',
+        'vso.test_write',
+        'vso.wiki_write',
+        'vso.work_full',
+        'vso.workitemsearch',
+    )
```

Comment:
Can you cut these down to just what we need for identity verification? Maybe no scopes initially? The identity pipeline usually just has scopes to obtain the users identity.

Reply:
@EvanPurkhiser The scopes must match the configuration setup in the application. Because this is the step users give the permissions to the application. And once set on VSTS side, there doesn't appear to be a way to alter them after the fact. So, rather than recreating the application each time I needed a scope change, I decided to just grab any scope we could reasonably want, and we could pair down after the functionality was decided upon. 

I'd say if you'd rather we went with the least possible permissions and redefine the app each time, then it definitely shouldn't be a part of this PR. I have a feeling what you might be thinking of (having two different sets of scopes for identity and integration) is not possible.

Reply:
> The scopes must match the configuration setup in the application

Ah I see, in that case in the `identity_pipeline_config` you don't need to pass them again, it will just use the identity scopes if that config option is ommited

https://github.com/getsentry/sentry/blob/c62a7b251ee9fc840a81621374ba34bf46a7ceb5/src/sentry/identity/oauth2.py#L39-L40

Reply:
Oh gotcha! Thanks! :)